### PR TITLE
feat(theme): update collapse to use theme & variants

### PR DESCRIPTION
**Pull request by brandonlilly**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [x] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [x] Added name to OWNERS file for all new components
- [x] If adding a new component, add its export to the rollup config

---------
**Outline your feature or bug-fix below**

My code does...

### CODE REVIEW

File: src/Collapse/Collapse.component.tsx [hunk 1]
```diff
@@ -68,42 +157,42 @@ export const Collapse = ({
...
                 !hasBottomBorder && 'no-bottom-border',
                 className
             )}
         >
-            {/** accordionHandler is used by CollapseGroup to handle closing/opening a single Collapse component. */}
             <button
-                onClick={() =>
-                    accordionHandler ? accordionHandler() : toggleOpen(!open)
-                }
+                onClick={onClick || (() => toggleOpen(!open))}
```
Comment (brandonlilly) [inline src/Collapse/Collapse.component.tsx:196]: @zero0Halo Sorry, I had to try :) Renaming to `onClick` makes more sense to me. Imo, this prop should have much more applicability than just the accordion.

Comment (zero0Halo) [reply src/Collapse/Collapse.component.tsx:196]: I am still of the opinion that event names should not also be treated as a handler. I do like how you consolidated the logic for toggleOpen though.

Comment (sharkparty) [reply src/Collapse/Collapse.component.tsx:196]: I added some thoughts on this [here](https://github.com/RetailMeNot/anchor/pull/54#discussion_r296771321).

File: src/Collapse/Collapse.component.tsx [hunk 2]
```diff
@@ -21,36 +22,124 @@ interface CollapseProps {
...
 
-// Associates files from Collapse/Themes directory to specific theme names
-export const CollapseThemes: { [key: string]: FlattenSimpleInterpolation } = {
-    comfortable: ComfortableTheme,
-    compact: CompactTheme,
-    none: css``,
+const variants = {
+    comfortable: css`
+        padding: 1rem 2rem;
+        border-top: solid ${colors.silver.dark} 1px;
```
Comment (sharkparty) [inline src/Collapse/Collapse.component.tsx]: I think this should be:
```border-top: solid thin ${th.color('borders.dark')};```
But, I don't think you had all the colors defined when this PR was submitted. I've updated the root branch to have all that stuff. How should we attack updating this PR? Want me to handle that?

File: src/Collapse/Collapse.component.tsx [hunk 3]
```diff
@@ -27,36 +23,124 @@ interface CollapseProps {
...
 
-// Associates files from Collapse/Themes directory to specific theme names
-export const CollapseThemes: { [key: string]: FlattenSimpleInterpolation } = {
-    comfortable: ComfortableTheme,
-    compact: CompactTheme,
-    none: css``,
+const variants = {
+    comfortable: css`
+        padding: 1rem 2rem;
+        border-top: solid thin ${th.color('neutrals.silver.dark')};
```
Comment (sharkparty) [inline src/Collapse/Collapse.component.tsx]: Shouldn't this pull from the `border.dark` entry?

File: src/Collapse/Collapse.component.tsx [hunk 4]
```diff
@@ -27,36 +23,124 @@ interface CollapseProps {
...
+            }
+
+            li a {
+                display: block;
+                padding-left: 3.4375rem;
+                height: 2.6875rem;
+                line-height: 2.6875rem;
+                cursor: pointer;
+                font-size: 0.875rem;
+                color: #222;
```
Comment (sharkparty) [inline src/Collapse/Collapse.component.tsx:120]: We should use the `Typography` component and override the `color` prop of that component, but for now can you add a `TODO` on ln113 for this work?

**Review by sharkparty — APPROVED** [review]
LKGTM

**Pull request merged**